### PR TITLE
Levelup only has default export

### PR DESCRIPTION
**Pull request by martingraf4**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [levelup only has default export](https://github.com/Level/levelup/blob/e73f536a46d7b8b2fa3d9b5abd0ae6efca41653e/lib/levelup.js#L333)

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## levelup ([<kbd>unpkg</kbd>](https://unpkg.com/browse/levelup@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'levelup'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. errors
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 53249c62421197b0333eb08b9b14abd875225f0e
</p>

**Comment by martingraf4** [comment]
I used default export instead of `export =` to not have to remove the other exports which are used by other packages. If I understand it correctly, this could break without `esModuleInterop`, so if someone has a better idea, happy to change it.

**Comment by istherepie** [comment]
The issue is that the export statement of the declaration files does not match the javascript package. 

The original package uses:

```js
module.exports = LevelUP;
```

The equivalent declaration would be:

```ts
export = LevelUP;
```

Ref: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

The `export default` is compiled to `export.default` which does not match the original package. 
(even though it might locally work)

We should avoid using it, which is why there is a dtslint check for it. 
It should not be disabled if not absolutely necessary!

**Comment by istherepie** [comment]
I've opened a separate pull request: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59253
in order to resolve this issue: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/59192

File: types/levelup/index.d.ts [hunk 1]
```diff
@@ -167,8 +169,5 @@ export interface LevelUpChain<K = any, V = any> {
     write(): Promise<this>;
 }
 
-export const errors: LevelUpConstructor["errors"];
-
-export const LevelUp: LevelUpConstructor;
-
-export {};
+declare const levelup: LevelUpConstructor;
+export default levelup;
```
Comment (istherepie) [inline types/levelup/index.d.ts:173]: Ref: https://github.com/Level/levelup/blob/e73f536a46d7b8b2fa3d9b5abd0ae6efca41653e/lib/levelup.js#L333

Line 333:
```js
module.exports = LevelUP
```

In order to match the export statement, the `export =` syntax should be used. 
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

example: 
```ts
export default levelup
```

File: types/level-ttl/level-ttl-tests.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { LevelUp as levelup } from 'levelup';
+import levelup from 'levelup';
```
Comment (istherepie) [inline types/level-ttl/level-ttl-tests.ts:1]: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

Export: 
```ts
export = levelup
```

Import: 
```ts
import levelup = require("levelup")
```

File: types/levelup/levelup-tests.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { LevelUp as levelup } from 'levelup';
+import levelup from 'levelup';
```
Comment (istherepie) [inline types/levelup/levelup-tests.ts:1]: See previously mentioned comment regarding import.

File: types/levelup/tslint.json [hunk 4]
```diff
@@ -8,6 +8,10 @@
                     [
                         "NeedsExportEquals",
                         false
+                    ],
+                    [
+                        "NoDefaultExport",
```
Comment (istherepie) [inline types/levelup/tslint.json:13]: This should not be disabled unless there is a good reason for this. 
(not just for passing the CI tests)

**Review by istherepie — CHANGES_REQUESTED** [review]
Thank you for the effort!
However due to the incorrect export statement it brings the declaration further away from matching the javascript package. 

Also avoid disabling the listing rules unless there is a good reason to do so.

**Comment by martingraf4** [comment]
Closing this PR in favor of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59253

**Pull request closed without merging**